Makefile

```diff
@@ -1,5 +1,10 @@
 .PHONY: all binary dynbinary build cross help install manpages run shell test test-docker-py test-integration test-unit validate win
 
+ifneq "$(wildcard hack/makefiles/*.mk)" ""
+# allow extending the makefile by adding includes in the hack/makefiles/ directory
+include hack/makefiles/*.mk
+endif
+
```

Comment:
Actually you don't have to enclose the include within a conditional; it's easier to use
`-include hack/makefiles/*.mk`

From GNU makefile docs:

> If you want make to simply ignore a makefile which does not exist or cannot be remade, with no error message, use the -include directive instead of include, like this:
>
> `-include filenames…`
>
> This acts like include in every way except that there is no error (not even a warning) if any of the filenames (or any prerequisites of any of the filenames) do not exist or cannot be remade.
>
> For compatibility with some other make implementations, sinclude is another name for -include.

Reply:
Also, this needs to be at the end of the file, so that the included makefiles can overwrite (have a preference over) the targets listed above.

Reply:
Updated